fv/wireguard_test.go

```diff
@@ -348,6 +335,52 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported", []apiconfig.Data
 			Expect(rcvd[1]).NotTo(BeEmpty())
 			Expect(rcvd[1]).To(Equal(sent[0]))
 		})
+
+		It("between pod to pod should be encrypted using wg tunnel with egress policy applied", func() {
+			policy := api.NewGlobalNetworkPolicy()
+
+			policy.Name = "f01-egress-deny"
+			policy.Spec.Egress = []api.Rule{{Action: api.Deny}}
+			policy.Spec.Selector = fmt.Sprintf("name in { '%s', '%s'}", wls[0].Name, wls[1].Name)
+			_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			cc.ExpectNone(wls[0], wls[1])
+			cc.ExpectNone(wls[1], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying tunnelled packet count is non-zero")
+			for i := range felixes {
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numInTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numOutTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+			}
+
+			cc.ResetExpectations()
+
+			policy.Name = "f01-egress-allow"
+			policy.Spec.Egress = []api.Rule{{Action: api.Allow}}
+			policy.Spec.Selector = fmt.Sprintf("name in { '%s', '%s'}", wls[0].Name, wls[1].Name)
+			_, err = client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			cc.ExpectSome(wls[0], wls[1])
+			cc.ExpectSome(wls[1], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying tunnelled packet count is non-zero")
+			for i := range felixes {
+				Eventually(func() int {
```

Comment:
For this to be useful it needs to check counts before and after you run connectivity.

IIUC, the first connectivity could result in tunnel packets while the policy is being installed.

Reply:
I've added a ResetCounts to the tcpdump - maybe useful once I've checked in my changes

Reply:
If I am understanding this comment correctly, we need to make sure that the policy is installed correctly before testing the connectivity. Will it be a good/clearer approach -- if we wait a few seconds and then read the policy and compare it with the policy created? If it matches then test the connectivity?

Reply:
I updated the PR with suggested point above.